Neo4j.Driver/Neo4j.Driver/Internal/Protocol/BoltProtocolV1.cs

```diff
@@ -172,5 +176,21 @@ public void Reset(IConnection connection)
         {
             connection.Enqueue(ResetMessage.Reset, null);
         }
+        
+        private void AssertNullOrEmptyTransactionConfig(TransactionConfig txConfig)
```

Comment:
I'm not sure about this but would it ever be possible to run 3.4 and 3.5 mixed - hit this assertion in this mixed version environment? What about simply ignoring `txConfig` (and warning about it) when connected to `V2` and `V1` - also address this in docs?